TellCore/NativeMethods.cs

```diff
@@ -24,29 +194,29 @@ internal static class NativeMethods
         [return: MarshalAs(UnmanagedType.CustomMarshaler, MarshalCookie = "out", MarshalTypeRef = typeof(TelldusUtf8Marshaler))]
         internal static extern string tdGetModel(int deviceId);
 
-        [DllImport("TelldusCore.dll")]
+        [DllImport("TelldusCore.dll", BestFitMapping = false, ThrowOnUnmappableChar = true)]
         [return: MarshalAs(UnmanagedType.CustomMarshaler, MarshalCookie = "out", MarshalTypeRef = typeof(TelldusUtf8Marshaler))]
         internal static extern string tdGetDeviceParameter(
             int deviceId,
             [MarshalAs(UnmanagedType.CustomMarshaler, MarshalCookie = "in", MarshalTypeRef = typeof(TelldusUtf8Marshaler))] string name,
             [MarshalAs(UnmanagedType.CustomMarshaler, MarshalCookie = "in", MarshalTypeRef = typeof(TelldusUtf8Marshaler))] string defaultValue);
 
-        [DllImport("TelldusCore.dll")]
+        [DllImport("TelldusCore.dll", BestFitMapping = false, ThrowOnUnmappableChar = true)]
         internal static extern bool tdSetName(
             int deviceId,
             [MarshalAs(UnmanagedType.CustomMarshaler, MarshalCookie = "in", MarshalTypeRef = typeof(TelldusUtf8Marshaler))] string name);
 
-        [DllImport("TelldusCore.dll")]
+        [DllImport("TelldusCore.dll", BestFitMapping = false, ThrowOnUnmappableChar = true)]
         internal static extern bool tdSetProtocol(
             int deviceId,
             [MarshalAs(UnmanagedType.CustomMarshaler, MarshalCookie = "in", MarshalTypeRef = typeof(TelldusUtf8Marshaler))] string protocol);
 
-        [DllImport("TelldusCore.dll")]
+        [DllImport("TelldusCore.dll", BestFitMapping = false, ThrowOnUnmappableChar = true)]
         internal static extern bool tdSetModel(
             int deviceId,
             [MarshalAs(UnmanagedType.CustomMarshaler, MarshalCookie = "in", MarshalTypeRef = typeof(TelldusUtf8Marshaler))] string model);
 
-        [DllImport("TelldusCore.dll")]
+        [DllImport("TelldusCore.dll", BestFitMapping = false, ThrowOnUnmappableChar = true)]
```

Comment:
To be honest, I do not understand this completely either. The reason I included it is that FxCop complained about a security issue when the attributes were missing.